lib/languageFeatures/documentFormatting.ts

```diff
@@ -15,21 +16,27 @@ export async function handleDocumentFormatting(params: DocumentFormattingParams)
   const path = await promises.mkdtemp(tmpdir() + sep);
   const tmpFile = path + sep + 'beautify';
   await promises.writeFile(tmpFile, text);
-  const emacs_script_path = __dirname + '/../../../emacs-vhdl-formating-script.lisp';
+  const rootDir = getRootDirectory();
+  const emacsScripts = join(rootDir, 'emacs-vhdl-formating-script.lisp');
+  const emacsLoadPath = join(rootDir, 'emacs');
   const numSpaces = typeof params.options.tabSize === 'number' ? params.options.tabSize : 2;
-  await promisify(exec)(`emacs --batch --eval "(setq-default vhdl-basic-offset ${numSpaces})" -l ${emacs_script_path}   -f vhdl-batch-indent-region ${tmpFile}`);
-  // Emacs VHDL mode hard tabs seem to be broken.
-  // Replace spaces in beginning with tabs
-  let newText = await promises.readFile(tmpFile, { encoding: 'utf8' });
-  if (params.options.insertSpaces === false) {
-    const re = /^( +)/gm;
-    console.log(re);
-    newText = newText.replaceAll(re,
-      match => '\t'.repeat(Math.ceil(match.length / numSpaces))
-    );
+  try {
+    await promisify(exec)(`emacs --batch --eval "(setq-default vhdl-basic-offset ${numSpaces})" --eval "(setq load-path (cons (expand-file-name \\"${emacsLoadPath}\\") load-path))" -l ${emacsScripts} -f vhdl-batch-indent-region ${tmpFile}`);
+    // Emacs VHDL mode hard tabs seem to be broken.
+    // Replace spaces in beginning with tabs
+    let newText = await promises.readFile(tmpFile, { encoding: 'utf8' });
+    if (params.options.insertSpaces === false) {
+      const re = /^( +)/gm;
+      console.log(re);
+      newText = newText.replaceAll(re,
+        match => '\t'.repeat(Math.ceil(match.length / numSpaces))
+      );
+    }
+    return [{
+      range: Range.create(document.positionAt(0), document.positionAt(text.length)),
+      newText
+    }];
+  } catch (e) {
+    throw new ResponseError(-1, 'Install emacs for formatting to work.');
```

Comment:
Can we catch only the emacs does not exists warning?
And show the error message or sth in other cases?

Reply:
We now show a vscode error message when emacs is not installed and otherwise the default error handling is done. Is this fine?